### ISSUE TITLE
Extend equality to `SplineInterpolation` 

### ISSUE THREAD

**Issue by tlow22** [jipolanco/BSplineKit.jl#74]
Hello, thanks for the helpful package! Was doing some unit testing today when I found that the cause of failure was that the equality operation `==` on `SplineInterpolation` doesn't work as I expected it to. 

Example: 
```
using BSplineKit

a = [0.0, 1.0, 2.0, 3.0, 4.0, 6.0]
b = [0.0, 1.0, 3.0, 12.0, 30.0, 100.0]

fit1 = interpolate(a, b, BSplineOrder(3))
fit2 = interpolate(a, b, BSplineOrder(3))

fit1 == fit2 # yields false
```

I can make a small PR that basically extends the `==` operator: 

```
Base.:(==)(o1::SplineInterpolation, o2::SplineInterpolation) =
    all([getproperty(o1, p) == getproperty(o2, p) for p in propertynames(o1)])
```

**Comment by jipolanco** [comment]
Hi, thanks for catching that and for your kind words!

Feel free to make a small PR (or I can do it myself if you prefer). I think the best solution would be to actually reuse the `==` operator already defined for `Spline`s.

Noting that `SplineInterpolation` is a subtype of `SplineWrapper`, the following definition should do the job and be quite generic:

```julia
# In src/Splines/wrapper.jl
Base.:(==)(o1::SplineWrapper, o2::SplineWrapper) = spline(o1) == spline(o2)
```

**Comment by tlow22** [comment]
Ah a much better solution, I'll get on it then.

**Comment by jipolanco** [comment]
Closed by #75
